### PR TITLE
Add residential rolling trash cans

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Add residential rolling trash cans."

#### Purpose of change
Many cities and towns have residential curbside trash pickup, creating efficient waste management systems. However, this was not modeled within the game, and I decided to add it. Idea comes from [this Reddit post](https://www.reddit.com/r/cataclysmdda/comments/14k3po1/real_question_why_arent_outdoor_garbage_cans/?share_id=f5XAUeLtLKQrzceW33XFc&utm_content=2&utm_medium=ios_app&utm_name=ioscss&utm_source=share&utm_term=1)

#### Describe the solution
Add rolling trash cans, and allow them to spawn in a variety of homes and other residential areas. A house currently has an 80% chance of having one of these outside, and they can be rolled around by the player. They also have a chance of having some miscellaneous waste items inside.

#### Describe alternatives you've considered
Not doing this and letting everyone drive to the dump to drop off trash.

#### Testing
Went around a city, found some cans, rolled them around, and put some stuff inside.

#### Additional context
None at this time.

### CODE REVIEW

File: data/json/items/containers/generic.json [hunk 1]
```diff
@@ -43,5 +43,26 @@
...
     "color": "light_gray"
+  },
+  {
+    "id": "trashcan_large",
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "industrial trash can" },
+    "looks_like": "f_trashcan",
+    "description": "A really big trashcan in which you can store, you guessed it, trash, ready for the garbage truck to pick up.  Made out of thick plastic, this can was meant for household or industrial use, something you roll up to the curb once a week.",
+    "weight": "4535 g",
```
Comment (casswedson) [inline data/json/items/containers/generic.json:54]: the link provides a unit weight of 39lb, about 17.7kg

File: data/json/items/containers/generic.json [hunk 2]
```diff
@@ -43,5 +43,26 @@
...
+    "//1": "Weight taken from https://www.uline.com/Product/Detail/H-7938BL/Trash-Cans-with-Wheels/Uline-Trash-Can-with-Wheels-95-Gallon-Black?pricode=WB3072&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&msclkid=d3c8827d02691eb87066e72c65ff5790",
+    "volume": "400 L",
+    "longest_side": "114 cm",
+    "price": "150 USD",
+    "//2": "Based on above link.",
+    "pocket_data": [
+      { "pocket_type": "CONTAINER", "max_contains_volume": "38 L", "open_container": true, "rigid": true, "watertight": true }
```
Comment (casswedson) [inline data/json/items/containers/generic.json:61]: about 10% of the total volume is used for storage, in this case I would just give the pocket about 90% of the total volume for storage ~360L or so, also the link mentions the capacity 330lbs about 149kg, please I would love to have that

File: data/json/items/containers/generic.json [hunk 3]
```diff
@@ -43,5 +43,26 @@
...
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "industrial trash can" },
+    "looks_like": "f_trashcan",
+    "description": "A really big trashcan in which you can store, you guessed it, trash, ready for the garbage truck to pick up.  Made out of thick plastic, this can was meant for household or industrial use, something you roll up to the curb once a week.",
+    "weight": "4535 g",
+    "//1": "Weight taken from https://www.uline.com/Product/Detail/H-7938BL/Trash-Cans-with-Wheels/Uline-Trash-Can-with-Wheels-95-Gallon-Black?pricode=WB3072&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&msclkid=d3c8827d02691eb87066e72c65ff5790",
+    "volume": "400 L",
+    "longest_side": "114 cm",
+    "price": "150 USD",
```
Comment (casswedson) [inline data/json/items/containers/generic.json:58]: as much as I find trading industrial trash cans for money after the end of the world funny, do add a price post apocalypse

Comment (MNG-cataclysm) [reply data/json/items/containers/generic.json:58]: Do you think that large bins of this nature would increase in price, or substantially drop?

File: data/json/items/containers/generic.json [hunk 4]
```diff
@@ -43,5 +43,34 @@
...
+  },
+  {
+    "id": "trashcan_large",
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "industrial trash can" },
+    "looks_like": "f_trashcan",
+    "description": "A really big trashcan in which you can store, you guessed it, trash, ready for the garbage truck to pick up.  Made out of thick plastic, this can was meant for household or industrial use, something you roll up to the curb once a week.",
+    "weight": "17700 g",
+    "//1": "Weight taken from https://www.uline.com/Product/Detail/H-7938BL/Trash-Cans-with-Wheels/Uline-Trash-Can-with-Wheels-95-Gallon-Black?pricode=WB3072&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&utm_source=Bing&utm_medium=pla&utm_term=H-7938BL&utm_campaign=Facilities%2BMaintenance&msclkid=d3c8827d02691eb87066e72c65ff5790",
```
Comment (akrieger) [inline data/json/items/containers/generic.json:55]: Link says 33lb which is 15kg rounding up like 50g. I can quibble about whether all houses would have the largest size but this is better than nothing.

**Comment by Procyonae** [comment]
I missed this PR and it's a bit late now but rather than x/y based placement it would've been much much nicer for maintainability to add these to the house palette as a 1x1 nest

**Pull request merged**